packages/cadl-ranch/src/actions/server-test.ts

```diff
@@ -0,0 +1,172 @@
+      options: {
+        requestBody: this.mockApiDefinition.request.body,
+        files: this.mockApiDefinition.request.files,
+        config: this.getConfigObj(),
+      },
+    });
+
+    if (this.mockApiDefinition.response.status !== response.status) {
+      logger.error(`Status code mismatch for ${this.name} endpoint`);
+      logger.error(`Expected: ${this.mockApiDefinition.response.status} - Actual: ${response.status}`);
+      throw new Error(`Status code mismatch for ${this.name} endpoint`);
+    }
+    if (this.mockApiDefinition.response.body) {
+      if (this.mockApiDefinition.response.body.contentType === "application/xml") {
+        if (JSON.stringify(this.mockApiDefinition.response.body.rawContent) !== JSON.stringify(response.data)) {
+          logger.error(`Response data mismatch for ${this.name} endpoint`);
+          logger.error(`Expected: ${this.mockApiDefinition.response.body["rawContent"]} - Actual: ${response.data}`);
+          throw new Error(`Response data mismatch for ${this.name} endpoint`);
+        }
+      } else if (Buffer.isBuffer(this.mockApiDefinition.response.body.rawContent)) {
+        if (
+          this.mockApiDefinition.request.headers &&
+          this.mockApiDefinition.request.headers["accept"] === "application/json"
+        ) {
+          if (response.data.content !== this.mockApiDefinition.response.body.rawContent.toString("base64")) {
+            throw new Error(`Response data mismatch for ${this.name} endpoint`);
+          }
+        } else {
+          if (
+            uint8ArrayToString(response.data, "utf-8") !== this.mockApiDefinition.response.body.rawContent.toString()
+          ) {
+            throw new Error(`Response data mismatch for ${this.name} endpoint`);
+          }
+        }
+      } else if (this.mockApiDefinition.response.body.contentType === "text/plain") {
+        if (this.mockApiDefinition.response.body.rawContent !== response.data) {
+          logger.error(`Response data mismatch for ${this.name} endpoint`);
+          logger.error(`Expected: ${this.mockApiDefinition.response.body} - Actual: ${response.data}`);
+          throw new Error(`Response data mismatch for ${this.name} endpoint`);
+        }
+      } else {
+        const responseData = JSON.stringify(response.data);
+        if (this.mockApiDefinition.response.body.rawContent !== responseData.replace(this.serverBasePath, "")) {
+          logger.error(`Response data mismatch for ${this.name} endpoint`);
+          logger.error(`Expected: ${this.mockApiDefinition.response.body} - Actual: ${response.data}`);
+          throw new Error(`Response data mismatch for ${this.name} endpoint`);
+        }
+      }
+    }
+    if (this.mockApiDefinition.response.headers) {
+      for (const key in this.mockApiDefinition.response.headers) {
+        if (this.mockApiDefinition.response.headers[key] !== response.headers[key].replace(this.serverBasePath, "")) {
```

Comment:
Originally, in the `typespec` package, the line is `if (this.mockApiDefinition.response.headers[key] !== response.headers[key])`. With this PR, I have changed it to `if (this.mockApiDefinition.response.headers[key] !== response.headers[key].replace(this.serverBasePath, ""))`. Because, the paging/LRO endpoints have nextLink in headers which have the baseURL. I am replacing it with spaces so, it can be compared and the validation will be successful.